src/editors/containers/ProblemEditor/data/OLXParser.test.js

```diff
@@ -23,227 +23,275 @@ import {
+      it('should equal ProblemTypeKeys.ADVANCED', () => {
+        expect(problemType).toEqual(ProblemTypeKeys.ADVANCED);
+      });
+    });
+    describe('given a problem with both a text and numeric input', () => {
+      const problemType = numericAndTextInputOlxParser.getProblemType();
+      it('should equal ProblemTypeKeys.ADVANCED', () => {
+        expect(problemType).toEqual(ProblemTypeKeys.ADVANCED);
+      });
+    });
   });
-  test('Test Advanced Problem Type by script tag', () => {
-    const olxparser = new OLXParser(scriptProblemOlX.rawOLX);
-    expect(() => olxparser.parseQuestions('numericalresponse')).toThrow(new Error('Script Tag, reverting to Advanced Editor'));
+  describe('getHints()', () => {
+    describe('given a problem with no hints', () => {
+      const hints = labelDescriptionQuestionOlxParser.getHints();
+      it('should return an empty array', () => {
+        expect(hints).toEqual([]);
+      });
+    });
+    describe('given checkbox olx with feedback and hints', () => {
+      const hints = checkboxOlxParser.getHints();
+      it('should equal an array of hints', () => {
+        expect(hints).toEqual(checkboxesOLXWithFeedbackAndHintsOLX.hints);
+      });
+    });
+    describe('given numeric olx with feedback and hints', () => {
+      const hints = numericOlxParser.getHints();
+      it('should equal an array of hints', () => {
+        expect(hints).toEqual(numericInputWithFeedbackAndHintsOLX.hints);
+      });
+    });
+    describe('given dropdown olx with feedback and hints', () => {
+      const hints = dropdownOlxParser.getHints();
+      it('should equal an array of hints', () => {
+        expect(hints).toEqual(dropdownOLXWithFeedbackAndHintsOLX.hints);
+      });
+    });
+    describe('given multiple choice olx with feedback and hints', () => {
+      const hints = multipleChoiceOlxParser.getHints();
+      it('should equal an array of hints', () => {
+        expect(hints).toEqual(multipleChoiceWithFeedbackAndHintsOLX.hints);
+      });
+    });
+    describe('given text input olx with feedback and hints', () => {
+      const hints = textInputOlxParser.getHints();
+      it('should equal an array of hints', () => {
+        expect(hints).toEqual(textInputWithFeedbackAndHintsOLX.hints);
+      });
+    });
   });
-  test('Test OLX with no question content should have empty string for question', () => {
-    const olxparser = new OLXParser(blankQuestionOLX.rawOLX);
-    const problemType = olxparser.getProblemType();
-    const question = olxparser.parseQuestions(problemType);
-    expect(question).toBe(blankQuestionOLX.question);
+  describe('parseMultipleChoiceAnswers()', () => {
+    describe('given a problem with no answers', () => {
+      const { answers } = multipleChoiceWithoutAnswersOlxParser.parseMultipleChoiceAnswers(
+        'multiplechoiceresponse',
+        'choicegroup',
+        'choice',
+      );
+      it('should return an array of objects with length one', () => {
```

Comment:
Nit: this is really a nit, you don't need to fix this. just an observation.
this is just a feeling about the wording here. If I ask myself what we are testing, it is probably that the answers are correct, or that all the answers are extracted. So while testing for length 1 is important, probably the name of the test should describe more what we are testing for (what we expect). I may be scratching my head here: "If there are no multiple choice answers in the OLX, why is an answer returned?" so the test name could be for example, "should return a default answer". And this is just for the naming of the test, the test itself is great.